### DISCUSSION TITLE
calculate degenerate eigenstate

### DISCUSSION THREAD

**Discussion by Yuanningss** [QuSpin/QuSpin#529] [General]
Hi 

I am wondering how to return the two eigenstates which have the same eigenenergies. 
<img width="320" alt="eigenstate" src="https://user-images.githubusercontent.com/104153268/165035241-e50a61c5-faf9-44f1-ba36-129a5ca9f2ea.png">
Here I choose k =1, and it only returns to one of the eigenstate. but if I choose k =2, it seems like the results are inconsistent with what I obtain for k =1. How to resolve this ?

**Comment by mgbukov** [comment]
if you have degeneracy, the spectral theorem postulates that the eigenspaces can be spanned by any vectors within the degenerate manifold; in other words, there is no way to fix a specific direction, the solver just gives any superposition within the eigenspace. Hermitian matrix eigensolvers typically orthogonalize the eigenvectors, but again, there's infinitely many orthogonal eigenbases.

**Reply by JiahaoYao** [reply]
You can also refer to https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.eigsh.html
